addons/mail/models/mail_channel.py

```diff
@@ -470,7 +470,7 @@ def _channel_channel_notifications(self, partner_ids):
         for partner in self.env['res.partner'].browse(partner_ids):
             user_id = partner.user_ids and partner.user_ids[0] or False
             if user_id:
-                for channel_info in self.with_user(user_id).channel_info():
+                for channel_info in self.with_user(user_id).with_context(allowed_company_ids=user_id.company_ids.ids).channel_info():
```

Comment:
Nitpicking: could we rewrap that long line for readability?
For example:

```suggestion
                user_channels = self.with_user(user_id).with_context(
                    allowed_company_ids=user_id.company_ids.ids
                )
                for channel_info in user_channels.channel_info():
```

Reply:
done